### ISSUE TITLE
Migrate xray sampling from aws sdk for go v1  to v2

### ISSUE THREAD

**Issue by wangzlei** [aws/aws-xray-sdk-go#473]
XRay sampling is the fundamental module, it uses AWS SDK for Go v1 retrieve xray sampling rules from xray service. Since [AWS SDK for Go v1 announces end of support on July 31 2025](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/), X-Ray go SDK can use net/http insteand of AWS SDK for Go v1 for xray centralized sampling.

**Comment by krzysiek1507** [comment]
Hi! Is there any ETA for this migration?

**Comment by jufemaiz** [comment]
:eyes: 🍿

**Comment by bestbug456** [comment]
Hello,

There is any ETA for this issue to be addressed? I would like to add to the discussion that the aws-sdk-go v1 contain an unfixed CVE, this mean that each software that contain this library that it will be scan (even by AWS inspector) will result containing the CVE due to this import. Regardless that the CVE make sense while using the xray, the scan software will still report the CVE. 
Thanks

**Comment by jufemaiz** [comment]
```slack
/giphy tumbleweed
```

**Comment by jj22ee** [comment]
Not only does X-Ray use aws-sdk-go v1 for Sampling, but it also needs it to [support tracing aws-sdk-go v1](https://github.com/aws/aws-xray-sdk-go/blob/7ae28931182316fd0fa6960008a8c5edd1d82e71/xray/aws.go#L165-L171) Clients.

To get rid of the aws-sdk-go dependency entirely, X-Ray SDK needs to remove support for instrumenting go sdk v1. This will be a breaking change.

**Comment by jufemaiz** [comment]
@jj22ee TBC when v1 is no longer supported, then the decision for AWS (not yet communicated) would to do one of the following:

1. Add a v2 to *every go package* and have it explicitly used, while deprecating (or deleting) *every* v1
2. Add a v2 package for *every* non-v2 package currently out there, mapping the references
3. Just knife the use of aws-sdk-go usage at some breaking point release version and have their people deal with it

TBH it's one of the biggest problems with a good intention from the people behind Go who never really thought that far out in terms of adoption and where others have dealt with similar issues (for example, go.mod not being there from the start as a way to manage dependencies due to a view of "a little copying is better than a little dependency" mindset).

**Comment by jj22ee** [comment]
Removal of v1 of aws-sdk-go is released in:
https://github.com/aws/aws-xray-sdk-go/releases/tag/v2.0.0

**Comment by Gobd** [comment]
I think this doesn't work the way it was done, it needs to be in a v2 folder https://go.dev/blog/v2-go-modules

**Comment by jj22ee** [comment]
Thanks @Gobd, the steps needed to make a correct v2 release for Go was missed by us, we'll have to test and make a re-release.

**Comment by Gobd** [comment]
Yeah I always have trouble with versions beyond 1 as well it's weird :) No problems take your time

**Comment by jj22ee** [comment]
Release is made in: https://github.com/aws/aws-xray-sdk-go/releases/tag/v2.0.0
In v2 of X-Ray SDK for Go, AWS SDK v1 has been removed as well as the AWS SDK v1 instrumentation.

Note that for v2 of the AWS X-Ray SDK for Go, users will need to add `v2` as a suffix to the imports of `github.com/aws/aws-xray-sdk-go`
```
Before: `github.com/aws/aws-xray-sdk-go/xyz/...`
After: `github.com/aws/aws-xray-sdk-go/v2/xyz/...`
```

And in the `go.mod`, the X-Ray Go dependency should be updated as follows:
```
Before:
github.com/aws/aws-xray-sdk-go v1.8.5
After:
github.com/aws/aws-xray-sdk-go/v2 v2.0.0
```

We'll update our README accordingly.
